platform-sdk/docs/proposals/TSS-Ledger-Id/TSS-Ledger-Id.md

```diff
@@ -721,230 +739,212 @@ The following startup sequence will occur:
-   2. clear the maps of the previous active roster data.
-3. If there existed a previous active roster and the candidate roster is different from the new active roster, log an
-   error.
-4. Set the new active roster
-5. If the active roster has private shares set, and there exists a candidate roster,
-   then `createTssMessageTransactions`.
-
-`On Candidate Roster`:
-
-1. If the internal candidate roster is set and equal to the input candidate roster, do nothing.
-2. Otherwise, set the internal candidate roster.
-3. If the active roster has private shares set, and there exists a candidate roster,
-   then `createTssMessageTransactions`.
-
-`On Private Shares`:
+   - Send the `TssVoteTransaction` to the `TssCryptographyManager`.
+4. Method `setCandidateRoster(Roster candidateRoster)`:
+   - If there already exists a candidate roster in the state, erase it and its related data from the `TssMessageMap`
+     and `TssVoteMap`.
+   - Set the new candidate roster in the state.
+   - Send the candidate roster to the `TssCryptographyManager`.
 
-1. Record the private shares in the internal private shares map for the appropriate roster hash.
-2. If the active roster has private shares set, and there exists a candidate roster,
-   then `createTssMessageTransactions`.
+#### `TssCryptographyManager`
 
-`createTssMessageTransactions`:
+The TSS manager executes on a separate thread and is responsible for all asynchronous cryptography computations that
+cannot be performed on the transaction handling thread. The computational responsibilities include:
 
-1. For each private share belonging to the active roster:
-   1. Use the TSS-Library to generate a TSS Message for the share private key and public keys for the candidate roster.
-   2. Submit a system `TssMessageTransaction` with the TSS Message.
+- Keying the candidate roster.
+- Submitting TssMessageTransactions to the transaction pool.
+- Submitting TssVoteTransactions to the transaction pool.
+- Computing public and private keys of shares when there is enough key material generated for the candidate roster.
+- Computing the ledger id.
+- Submitting the candidate roster to the platform for adoption when the candidate roster has enough key material to
+  generate the ledger id.
 
-#### TSS Message Validator
-
-The TSS message validator is responsible for validating TSS messages and for submitting votes as to the validity of the
-candidate roster. The TSS message validator gets its input from the TSS state manager in the form of a candidate roster
-that we want to adopt, and as a sequence of TSS messages that have reached consensus.
+##### `TssCryptographyManager` Details
 
 Internal Elements:
 
-1. active roster hash
-2. candidate roster hash
-3. Map<RosterHash, Roster> rosters
-4. Map<RosterHash, List<TssMessageTransaction>> tssMessages
-5. Map<RosterHash, List<TssVoteTransaction>> tssVotes
-6. Set<RosterHash> votingClosed
-7. private RSA signing key (constructor argument)
+1. NodeId (constructor)
```

Comment:
This and the below seem to be extremely low-level details of the implementation. While I'm curious to learn about them, I'm unsure if this belongs to the design doc.

Reply:
When everything was separated into multiple components, it was essential to see how the components were communicating and interacting.  Now that everything is in a single component, I can agree that the details may be unnecessary at this point.  But I wasn't able to evaluate the multi-component design without the details of who communicates to who and who needs what in order to perform their function.  

I don't want to delete it.   I think its still relevant details to see what is necessary as input and what is generated as output when.